spec/e2e/enum_spec.rb

```diff
@@ -68,6 +82,31 @@
       expect(StoredFile.as(:f).where(id: file.id).pluck('f.flag')).to eq([nil])
       expect(file.reload.flag).to eq(nil)
     end
+
+    it "raises error if value doesn't match an enum key" do
+      file = StoredFile.new
+      file.type = :audio
+      expect { file.save! }.to raise_error Neo4j::Shared::Enum::InvalidEnumValueError
+    end
+
+    it 'respects local _case_sensitive option' do
+      file = StoredFile.new
+      file.type_format = :png
+      expect { file.save! }.to raise_error(Neo4j::Shared::Enum::InvalidEnumValueError)
```

Comment:
For a stronger assertion this could be:

```ruby
expect { file.save! }.to raise_error(Neo4j::Shared::Enum::InvalidEnumValueError, 'Value passed to an enum property must match one of the enum keys')
```